format/proto/types.proto

```diff
@@ -28,13 +28,35 @@ message Metadata {
     optional sfixed32 length = 2;
 }
 
-//  NodeMetrics is composed of the various metrics of a node
+//  ClusterView is a node's view of the cluster.
+//  This is sent in response to a heartbeat request by a
+//  Management Server.
+//  It stores its own endpoint and server metrics and also
+//  the network condition of all its peers.
+message ClusterView {
+    // Current node's Endpoint
+    optional string endpoint = 1;
+    // Current node's Server Metrics
+    optional ServerMetrics serverMetrics = 2;
+
+    // Node's view of peer's connectivity health
+    map<string, NetworkMetrics> networkMetricsMap = 3;
+}
+
+//  NetworkMetrics contains the network metrics of a node:
+//  whether the last heartbeat was responded or not,
+//  heartbeat response time and polling success rate.
+message NetworkMetrics {
+    optional bool lastHeartbeatStatus = 1;
+    optional int32 heartbeatResponseTime = 2;
+    optional int32 pollSuccessRate = 3;
+}
+
+//  ServerMetrics is composed of the various metrics of a node
 //  like the status of the different components, pending
 //  requests in the log unit and tokens generated by the
 //  sequencer on the node.
-//  This is created in response to a heartbeat request by a
-//  Management Server.
-message NodeMetrics {
+message ServerMetrics {
```

Comment:
Ok I was wondering if we would ever need UNKNOWN.
For now I can replace this with a simple enum with these 3 options. UP, DOWN and UNKNOWN.